src/Database/FunctionsBuilder.php

```diff
@@ -158,6 +158,30 @@ public function coalesce($args, $types = [])
         return $this->_build('COALESCE', $args, $types, current($types) ?: 'string');
     }
 
+    /**
+     * Returns a FunctionExpression representing a call to SQL CAST function.
+     *
+     * @param string|\Cake\Database\ExpressionInterface $field Field or expression to cast.
+     * @param string $type The target data type
+     * @return \Cake\Database\Expression\FunctionExpression
+     */
+    public function cast($field, $type = '')
+    {
+        if (is_array($field)) {
+            deprecationWarning(
+                'Build cast function by FunctionsBuilder::cast(array $args) is deprecated. ' .
+                'Use FunctionsBuilder::cast($field, string $type) instead.'
+            );
+
+            return $this->_build('CAST', $field);
+        }
+
+        $expression = $this->_literalArgumentFunction('CAST', $field);
```

Comment:
Shouldn't this set the return type to $type ?

Reply:
No, because $type is SQL type.

Reply:
Well the return type defaults to string. We should probably provide a way to set it.

Reply:
I didn't found where the return type used. If it's required, I will add an arguments to set it.